src/libs/utils.js

```diff
@@ -369,6 +369,29 @@ export const truncateInteger = integer => {
   return `${Math.floor(integer / 1000)}k`
 }
 
+/**
+ * Polls using a given function until the pollUntil function returns true.
+ * @param {() => { result: A, shouldContinue: boolean }} pollFn - The function to poll using
+ * @param {number} pollTime - How much time there should be in ms between calls of the pollFn
+ * @param {boolean} leading - Whether the function should wait {pollTime} ms before running for the first time
+
+ * @returns {A} - The response of pollFn
+ */
+export const poll = async (pollFn, pollTime, leading = true) => {
+  do {
+    !leading && await delay(pollTime)
+    const r = await pollFn()
+    if (!r.shouldContinue) { return r.result }
+    leading && await delay(pollTime)
+  } while (true)
```

Comment:
Possible improvement
```suggestion
    leading = false
  } while (r.shouldContinue)
  return r.result
```
Modifying `leading` avoids the need for the early return and avoids the repeated fragment `await delay(pollTime)`. This is at the cost of making `leading` mutable; it seems worth it to me though.

Reply:
I like the leading change. I am not going to change the while though, because it involves pulling the `r` out of the loop, and I like the immutability of setting it within the loop.